### PR TITLE
Updating buildpack scripts

### PR DESCRIPTION
**Pull request by debabrata-shome**
## What does the PR Do?

-  Adding error handling for download failure in lib/failure.sh file.
-  Updating bin/compile file.
-  Updating stdout error message.

## GUS Work Item 
[W-9257549](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07AH000000LYgZYAW/view)

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -91,51 +91,52 @@ meta_setup
 # the "build-step" key keeps track of where we are in the execution
 # so that we can see where failed builds errored
 meta_set "build-step" "pre-init"
+# Downloading the file using curl command.
+make_a_curl_call(){
```
Comment (LokeshNSF) [inline bin/compile:95]: let this function name be download_config_files()

File: bin/compile [hunk 2]
```diff
@@ -91,51 +91,52 @@ meta_setup
...
-
-  echo "Downloading config files for the Sfdocs release tag: $release_tag"
-  local package_json="https://api.github.com/repos/sfdocs/sfdocs-repo-build-config/contents/package.json"
-  local yarn_lock="https://api.github.com/repos/sfdocs/sfdocs-repo-build-config/contents/yarn.lock"
-
-  local package_json_url=$(get_file_url "$release_tag"  "$package_json")
-  local yarn_lock_url=$(get_file_url "$release_tag" "$yarn_lock")
-
-  local code=$(curl "$package_json_url" -L --silent --fail --retry 5 --retry-max-time 15 -H "Authorization: token $SFDOCS_CONFIG_TOKEN" -H "Accept: application/vnd.github.v3.raw" -o "$BUILD_DIR/package.json" --write-out "%{http_code}")
+  local curl_url=$(get_file_url "${file_url}")
```
Comment (LokeshNSF) [inline bin/compile:99]: local file_url.

File: bin/compile [hunk 3]
```diff
@@ -91,51 +91,52 @@ meta_setup
...
+    return
+  fi
+  # Removing package.json and yarn.lock that comes from repo before downloading 
+  # it from sfdocs-repo-build-config
+  rm -f $BUILD_DIR/package.json $BUILD_DIR/yarn.lock
+
+  echo "Downloading config files for the Sfdocs release tag: ${SFDOCS_RELEASE_TAG}"
+  for item in "${files_to_download[@]}"
+  do
+    file_name=$(echo "${item}"|awk -F ":" '{print $1}'| xargs)
```
Comment (LokeshNSF) [inline bin/compile:136]: Complicated man :)
Can we use the awk and get key and value in one step itself and assign them to file_name and file_url?

File: bin/compile (hunk 1 above)
Comment (debabrata-shome) [reply bin/compile]: Updated method name

File: bin/compile (hunk 2 above)
Comment (debabrata-shome) [reply bin/compile]: updated variable name

File: bin/compile (hunk 3 above)
Comment (debabrata-shome) [reply bin/compile]: As per our offline discussion, updated the download logic

**Review by LokeshNSF — APPROVED** [review]
LGTM!.

**Comment by debabrata-shome** [comment]
@LokeshNSF  approved this PR. Merged the PR.

**Pull request merged**